Util/src/main/java/io/deephaven/util/compare/BooleanComparisons.java

```diff
@@ -0,0 +1,89 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.util.compare;
+
+public class BooleanComparisons {
```

Comment:
Why do we need this? We don't ever allow a column to contain primitive booleans.

Reply:
This allows us to not have to special case some boolean code generation cases; namely, `BooleanChunkEquals` and `BooleanChunkHasher` will now call this class (just like the other ChunkEquals / ChunkHasher impls). Maybe we can make the argument that `BooleanComparisons` should only have `eq` and `hashCode`, and not `compare`, `gt`, etc...?